tiledb/sm/array_schema/test/unit_dimension.cc

```diff
@@ -38,6 +38,63 @@ using namespace tiledb;
 using namespace tiledb::common;
 using namespace tiledb::sm;
 
+using Datatype = tiledb::sm::Datatype;
+
+template <class T>
+struct type_to_datatype {
+  static Datatype datatype;
+};
+
+template <>
+struct type_to_datatype<int8_t> {
```

Comment:
This can be done with `to_underlying` in `tiledb/stdx/utility/to_underlying.h`: https://github.com/TileDB-Inc/TileDB/blob/dev/tiledb/stdx/utility/to_underlying.h

  - see usage: https://github.com/TileDB-Inc/TileDB/blob/dev/tiledb/sm/enums/datatype.h#L369

Reply:
(these definitions also appear to duplicate the `RangeTraits` definitions in `test/src/helpers-dimension.h`: https://github.com/TileDB-Inc/TileDB/blob/dev/test/src/helpers-dimension.h#L67-L102)

Reply:
@ihnorton to_underlying seems to be working on the enum as an input... I'm trying to do the opposite, get the enum value from the underlying type.

The definitions in RangeTraits are in the test branch. This is in the unit test branch. I'm not sure what a common location would be for these.